source/extensions/stat_sinks/common/statsd/tag_formats.h

```diff
@@ -0,0 +1,40 @@
+#pragma once
+
+#include <string>
+
+namespace Envoy {
+namespace Extensions {
+namespace StatSinks {
+namespace Common {
+namespace Statsd {
+
+enum class TagPosition { TagAfterValue, TagAfterName };
+
+struct TagFormat {
+  const std::string& start;
+  const std::string& assign;
+  const std::string& separator;
+  const TagPosition tag_position;
+};
+
+// TODO: construct on first use?
```

Comment:
Yes please, same below.

Reply:
Tried to use `CONSTRUCT_ON_FIRST_USE` here but for some reason the resulting code couldn't compile. Interestingly, expanding the macro manually did help. I've saved the error message and code [as a gist](https://gist.github.com/vaccarium/6409c255efaeabc23e5f547f7bcad19a), could someone more familiar with the codebase please take a look?